Src/DependencyCollector/Shared/SanitizedHostList.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Globalization;
+
+    /// <summary>
+    /// Sanitized collection on host strings.
+    /// </summary>
+    internal class SanitizedHostList : ICollection<string>
+    {
+        private List<string> hostList = new List<string>();
+
+        #region ICollection Implemenation
+        public int Count
+        {
+            get
+            {
+                return this.hostList.Count;
+            }
+        }
+
+        public bool IsReadOnly
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// We sanitize before adding to the list. We try our best to extract the host name from the passed in item and store that in the collection.
+        /// </summary>
+        /// <param name="item">Item to be added.</param>
+        public void Add(string item)
+        {
+            if (string.IsNullOrEmpty(item))
+            {
+                // Since this is called in the context of the config. If there is an empty <Add /> element, we wil just ignore and move on.
+                return;
+            }
+
+            Uri uriResult;
+            if (Uri.TryCreate(item, UriKind.Absolute, out uriResult))
+            {
+                this.AddIfNotExist(uriResult.Host);
+                return;
+            }
+            else if (Uri.TryCreate("http://" + item, UriKind.Absolute, out uriResult))
```

Comment:
I do not understand why you need this complicated logic. Can we just have a list of strings and compare the host by `string.Contains` instead?